http.js

```diff
@@ -0,0 +1,41 @@
+//------------------------------------------------------------------------------
+// HTTP helpers
+//------------------------------------------------------------------------------
+const http = require('http');
+var Q = require("q");
+
+function fetchURL(url){
+    var deferred = Q.defer();
+    http.get(url, (res) => {
+        var body = "";
+        res.on('data', function (chunk) {
+            body += chunk;
+        });
+        res.on('end', () => {
+            deferred.resolve({'response': res, "body": body});
+        });
+    }).on('error', (e) => {
+        console.error(JSON.stringify(e));
+        deferred.reject("failed to get a response from url: " + url);
+    });
+    return deferred.promise;
+}
+function fetchURLIntoJSON(url){
```

Comment:
if we ever need a new version of this function, say `fetchURLIntoXML` or something, we should just make it driven by an encoding.

`fetchURL(url, [encoding])` where encoding is text, json, XML etc.

Just food for thought. This seems more in line with similar apis.